### PR TITLE
CORE-5553 token record creation

### PR DESCRIPTION
**Pull request by lankydan**

### CODE REVIEW

**Comment by ronanbrowne** [comment]
FYI target branch needs to be changed to `release/os/5.0-beta1` (and subsequently then this should be forward merged to `release/os/5.0` )

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoPersistTransactionRequestHandler.kt [hunk 1]
```diff
@@ -11,13 +11,16 @@ import net.corda.v5.base.util.contextLogger
...
-class UtxoPersistTransactionRequestHandler(
+
+class UtxoPersistTransactionRequestHandler @Suppress("LongParameterList") constructor(
+    private val holdingIdentity: HoldingIdentity,
     private val transaction: UtxoTransactionReader,
     private val tokenObservers: UtxoTokenObserverMap,
     private val externalEventContext: ExternalEventContext,
     private val persistenceService: UtxoPersistenceService,
-    private val utxoOutputRecordFactory: UtxoOutputRecordFactory
+    private val utxoOutputRecordFactory: UtxoOutputRecordFactory,
```
Comment (blsemo) [inline components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoPersistTransactionRequestHandler.kt]: Why'd you add a comma here?

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoTransactionReaderImpl.kt [hunk 2]
```diff
@@ -60,7 +65,25 @@ class UtxoTransactionReaderImpl(
     override val relevantStatesIndexes: List<Int>
         get() = transaction.relevantStatesIndexes ?: emptyList()
 
-    override fun getProducedStates(): List<StateAndRef<ContractState>> = wrappedWireTransaction.outputStateAndRefs
+    override fun getProducedStates(): List<StateAndRef<ContractState>> {
```
Comment (blsemo) [inline components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoTransactionReaderImpl.kt:67]: Couldn't you just have done a `wrappedWireTransaction.ouputStateAndRefs.withIndex().filter(indexed -> indexed.index in relevantStateIndexes)`?

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoTransactionReaderImpl.kt [hunk 3]
```diff
@@ -60,7 +65,25 @@ class UtxoTransactionReaderImpl(
...
+        return rawGroupLists[UtxoComponentGroup.OUTPUTS.ordinal]
+            .zip(rawGroupLists[UtxoComponentGroup.OUTPUTS_INFO.ordinal])
+            .withIndex()
+            .filter { indexed -> indexed.index !in relevantStatesIndexes }
```
Comment (blsemo) [inline components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoTransactionReaderImpl.kt]: is that filter statement correct? Wouldn't that filter out the relevant states and just add the irrelevant ones?

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoPersistTransactionRequestHandler.kt (hunk 1 above)
Comment (owenstanford) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoPersistTransactionRequestHandler.kt]: removed

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoTransactionReaderImpl.kt (hunk 3 above)
Comment (owenstanford) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoTransactionReaderImpl.kt]: no, fixed

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoTransactionReaderImpl.kt (hunk 2 above)
Comment (owenstanford) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoTransactionReaderImpl.kt:67]: maybe, I've left it as-is as it's working atm.

**Review by blsemo — APPROVED** [review]
Ok - can you raise a tech debt ticket for using WrappedWireTransaction? I'd prefer to not duplicate the code to built `StateAndRef` instances.

**Pull request merged**